### PR TITLE
docs: added comment about merge queues

### PR DESCRIPTION
**Pull request by dougch**
### Resolved issues:

None

### Description of changes: 

This is a test of merge queues.

### Call-outs:

This repo isn't a great candidate for merge-queues and we might not want to merge this.

### Testing:

Is this a refactor change? If so, how have you proved that the intended behavior hasn't changed? -->

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ To send us a pull request, please:
 4. Commit to your fork using clear commit messages.
 5. Send us a pull request, answering any default questions in the pull request interface.
 6. Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
+7. Once CI passes, you'll need to add your change to a [merge-queue](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request-with-a-merge-queue), which will run CI again with any other queue up PRs.  The merge will happen 
+automatically or if there are failures, the PR will be evicted from the queue.
```
Comment (lrstewart) [inline CONTRIBUTING.md:39]: Nit: I don't think we generally manually newline in markdown, and the rest of this paragraph definitely doesn't.
```suggestion
7. Once CI passes, you'll need to add your change to a [merge-queue](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request-with-a-merge-queue), which will run CI again with any other queue up PRs.  The merge will happen automatically or if there are failures, the PR will be evicted from the queue.
```

**Review by lrstewart — DISMISSED** [review]

**Comment by dougch** [comment]
Missed updating https://github.com/aws/s2n-netbench/blob/main/.github/workflows/ci.yml#L308

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -129,14 +129,14 @@ jobs:
...
-        if: github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name
+        if: ( github.event_name == 'merge_group' || github.event_name == 'push' ) || github.repository == github.event.pull_request.head.repo.full_name
         with:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: us-west-1
 
       - name: Upload to S3
-        if: github.event_name == 'push' || github.repository == github.event.pull_request.head.repo.full_name
+        if: ( github.event_name == 'merge_group' || github.event_name == 'push' ) || github.repository == github.event.pull_request.head.repo.full_name
```
Comment (lrstewart) [inline .github/workflows/ci.yml:139]: Is "or" correct here? Will this publish the report twice, or is the event not a "push" from a merge group?

Comment (dougch) [reply .github/workflows/ci.yml:139]: Correct, the event from `merge_group` is called `check_requested`, so this runs once when the merge queue is setup and then again on the final push to `main`.

**Review by lrstewart — APPROVED** [review]

**Pull request merged**